pkg/services/store/service.go

```diff
@@ -86,3 +108,82 @@ func (s *standardStorageService) Read(ctx context.Context, user *models.SignedIn
 	// TODO: permission check!
 	return s.tree.GetFile(ctx, path)
 }
+
+func isFileTypeValid(filetype string) bool {
+	if (filetype == "image/jpeg") || (filetype == "image/jpg") || (filetype == "image/gif") || (filetype == "image/png") || (filetype == "image/svg+xml") || (filetype == "image/webp") {
+		return true
+	}
+	return false
+}
+
+func (s *standardStorageService) Upload(ctx context.Context, user *models.SignedInUser, form *multipart.Form) (*Response, error) {
+	response := Response{
+		path:       "upload",
+		statusCode: 200,
+		message:    "Uploaded successfully",
+		err:        false,
+	}
+	upload, _ := s.tree.getRoot("upload")
+	if upload == nil {
+		response.statusCode = 404
+		response.message = "upload feature is not enabled"
+		response.err = true
+		return &response, fmt.Errorf("upload feature is not enabled")
+	}
+
+	files := form.File["file"]
+	for _, fileHeader := range files {
+		// Restrict the size of each uploaded file to 1MB.
+		if fileHeader.Size > MAX_UPLOAD_SIZE {
+			response.statusCode = 400
+			response.message = "The uploaded image is too big"
+			response.err = true
+			return &response, nil
+		}
+
+		// open each file to copy contents
+		file, err := fileHeader.Open()
+		if err != nil {
+			return nil, err
+		}
+		err = file.Close()
+		if err != nil {
+			return nil, err
+		}
+		data, err := ioutil.ReadAll(file)
+		if err != nil {
+			return nil, err
+		}
+		filetype := http.DetectContentType(data)
+		// only allow images to be uploaded
+		if !isFileTypeValid(filetype) && !strings.HasSuffix(fileHeader.Filename, ".svg") {
```

Comment:
Sorry – maybe too late here. Any file with `.svg` ext may be uploaded?

Reply:
I suppose yes, we should not rely only on extension – but need to figure out the best way. I think on a meeting you said that `http.DetectContentType` does not work for us – but I missed the root cause, could you please elaborate more? 

Also, I think I am missing the idea what we really aim to allow uploading - any valid image and any valid svg?

Reply:
> I think on a meeting you said that http.DetectContentType does not work for us
When I uploaded some svg files I had locally, the content type was detected as `text/xml; charset=utf-8`. This means it won't show up on the FE as the component expect "image/svg+xml" to render svg file. 
> I think I am missing the idea what we really aim to allow uploading - any valid image and any valid svg?
We have 2 different use cases for uploading right now: 
1. uploading icon in svg format - so allowing any valid svg
2. uploading background image in some (almost all?) image types (jpg, jpeg, webp, gif, png)